### DISCUSSION TITLE
❓ Question: How to load RAW audio instead of only WAV?

### DISCUSSION THREAD

**Discussion by hobojoe** [snakers4/silero-vad#261] [Q&A] [answered]
## ❓ Questions and Help

Hi,

First of all, congratulations on the VAD model, it is great!

I have a question though: Can I load RAW audio files instead of the WAV?

I am currently processing streaming audio in RAW format: (ALAW - Mono - 8khz). Since the volume of data is high, I would like to avoid the conversion to WAV before running the model.

Thanks!

**Comment by adamnsandle** [comment]
Could you upload an example of a single alaw file?

**Comment by hobojoe** [comment]
> Could you upload an example of a single alaw file?

Sure:
https://www.dropbox.com/s/s9may9l58szafb0/01b084d5-e5e3-4348-b14e-beee32cb6909.raw?dl=1

**Comment by adamnsandle** [comment, marked as answer]
I was able to load your example using the following code:
```python3
import soundfile as sf
wav, sr = sf.read('files/01b084d5-e5e3-4348-b14e-beee32cb6909.raw', samplerate=8000, channels=1, subtype='ALAW', dtype='float32')
wav = torch.tensor(wav)
```
Then you can use VAD model to process this chunk.
For example:
```python3
## just probabilities

speech_probs = []
window_size_samples = 256
for i in range(0, len(wav), window_size_samples):
    chunk = wav[i: i+window_size_samples]
    if len(chunk) < window_size_samples:
        break
    speech_prob = model(chunk, 8000).item()
    speech_probs.append(speech_prob)
model.reset_states() # reset model states after each audio

print(speech_probs[:10]) # first 10 chunks predicts
```
